langchain, community: Implement Ontotext GraphDB QA Chain

nelly-hateva opened the pull request:
- **Description:** Implement Ontotext GraphDB QA Chain
  - **Issue:** N/A
  - **Dependencies:** N/A
  - **Twitter handle:** @OntotextGraphDB

nelly-hateva committed:
SOL-42: GraphDB QA Chain

nelly-hateva committed:
Merge remote-tracking branch 'origin' into GraphDBQAChain

nelly-hateva committed:
Format, lint & spell check fixes

nelly-hateva committed:
Rename repository

nelly-hateva committed:
Fix notebook imports

nelly-hateva committed:
Lint, format, import and code spell fixes

nelly-hateva committed:
Fix community extended tests

nelly-hateva committed:
Fix notebook formatting

nelly-hateva committed:
Documentation fixes

nelly-hateva committed:
Merge remote-tracking branch 'origin' into GraphDBQAChain

nelly-hateva commented:
I've synced this branch with the master branch. @baskaryan, @eyurtsev, @hwchase17, please, approve the workflows again and provide feedback. Is there anything else I should do, so this PR is merged?

baskaryan reviewed (commented):
sorry for delay @nelly-hateva!

35k lines is pretty big, could we reduce the size of the various example files?

also would it be clearer to name classes OntoTextGraphDB... instead of GraphDB...? GraphDB sounds like an abstraction more than a specific integration, may confuse folks

nelly-hateva committed:
Rename GraphDB to Ontotext GraphDB; Delete test files and instead generate ones in the tests

nelly-hateva committed:
Merge remote-tracking branch 'origin/master' into GraphDBQAChain

nelly-hateva committed:
Remove key commited by mistake

nelly-hateva committed:
Add support for explicit rdf format for the local file

nelly-hateva committed:
Format, linting

nelly-hateva commented:
Hello @baskaryan . Thank you for the quick reply. I completely agree with you and I've deleted all example files and heavily reduced the size of the changes. I've also renamed classes containing "GraphDB" to "OntotextGraphDB" (note that you proposed "OntoTextGraphDB", but "Ontotext" is a company name and it's written with lowercase "t") and added links to the website for clarity in the classes docs. Please, approve the workflows.

baskaryan commented on docs/vercel.json:
```diff
@@ -1444,6 +1444,10 @@
       "source": "/en/latest/modules/chains/examples/graph_cypher_qa.html",
       "destination": "/docs/use_cases/graph/graph_cypher_qa"
     },
+    {
+      "source": "/en/latest/modules/chains/examples/graph_ontotext_graphdb_qa.html",
+      "destination": "/docs/use_cases/graph/graph_ontotext_graphdb_qa"
+    },
```
only needed for when we've moved doc pages around and wanted to not break old links. no need to do this for new docs

nelly-hateva replied:
Tnx. Fixed

baskaryan commented on libs/langchain/langchain/graphs/ontotext_graphdb_graph.py:
no need to add this, only needed for backwards compatibility for classes that used to live in libs/langchain and were moved

nelly-hateva replied:
Tnx. Removed. Do I need the changes in libs/langchain/langchain/graphs/__init__.py and in libs/langchain/tests/unit_tests/graphs/test_imports.py or not?

nelly-hateva committed:
Merge remote-tracking branch 'origin' into GraphDBQAChain

nelly-hateva committed:
Address MR comments & fix langchain IT

nelly-hateva committed:
Change prompt and notebook

nelly-hateva committed:
Except any error not just parse exception

baskaryan commented on libs/langchain/langchain/graphs/__init__.py:
```diff
@@ -35,4 +35,5 @@ def __getattr__(name: str) -> Any:
     "RdfGraph",
     "ArangoGraph",
     "FalkorDBGraph",
+    "OntotextGraphDBGraph",
```
delete

nelly-hateva replied:
Thank you. Removed

nelly-hateva committed:
MR comments fixes, fix prompt changes, updated documentation

nelly-hateva committed:
FIx notebook formatting

baskaryan commented:
awesome stuff, thank you @nelly-hateva!!
